rust/blockstore/src/arrow/blockfile.rs

```diff
@@ -319,11 +354,14 @@ impl<'me, K: ArrowReadableKey<'me> + Into<KeyWrapper>, V: ArrowReadableValue<'me
         let target_block_id = self.sparse_index.get_target_block_id(&search_key);
```

Comment:
I added a task in tracker to remove Box<dyn ChromaError> and instead add specific error types